neqo-http3/src/settings.rs

```diff
@@ -228,6 +251,13 @@ impl ZeroRttChecker for HttpZeroRttChecker {
                 let value = setting.value == 1;
                 self.settings.get_webtransport() || !value
             }
+            HSettingType::EnableH3Datagram => {
+                if setting.value > 0 {
+                    return false;
+                }
+                let value = setting.value == 1;
```

Comment:
So this is a cargo culting of previous code, but I think that it needs to be looked at more carefully.

What we want is in this truth table.

| ... | Old Enabled | Old Disabled |
|---|---|---|
| New Enabled | Enable 0-RTT | Enable 0-RTT |
| New Disabled | Disable 0-RTT | Enable 0-RTT |

That is, it is OK to resume unless the setting was previously enabled (`setting.value == 1`) and the setting is now disabled.

This matches what the code does, mostly.  This check:

```rust
                if setting.value > 0 {
                    return false;
                }
```

Is going to disable 0-RTT if the old connection enabled the feature.  I think that there is a very simple fix:

```suggestion
                if setting.value > 1 { // <--- here!
                    return false;
                }
                let value = setting.value == 1;
```

But this highlights that there might be a problem with our 0-RTT tests in that we weren't testing these cases properly.  Can you take a look to see if there are tests for this?

Reply:
Yes, we have a test [here](https://searchfox.org/mozilla-central/rev/188d0f76a73e0671d12e744a71e9f5701668cc37/third_party/rust/neqo-http3/src/features/extended_connect/tests/webtransport/negotiation.rs#84), but it doesn't really check `early_data_accepted`. I'll also try to fix this.